executor/README.md

```diff
@@ -0,0 +1,242 @@
+Cook Executor
+=============
+
+The cook executor is a custom executor written in python. It replaces the default command executor in order to enable a number of features for both operators and end users.
+
+For Users
+-----
+
+The cook executor supports a number of (completely optional) features beyond the standard Mesos command executor. It allows users to specify additional commands that run either before or after a job's primary command. This enables the use of existing utilities, and promotes sharing and reuse. It also provides an HTTP API that allows commands to expose progress updates and custom failure messages.
+
+### Scheduler HTTP API
+
+#### `POST /rawscheduler`
+
+The job creation endpoint can be used to specify before and after commands, with the following keys:
+
+| key | type | description |
+|-----|------|-------------|
+| `before_commands` | array | An array of `command`s to run before all user specified commands. |
+| `after_commands` | array | An array of `command`s to run after all user specified commands. |
+
+Commands have the following structure:
+
+| key | type | description |
+|-----|------|-------------|
+| `value` | string | A string representing the command to be run, ex: `echo hello`. |
+| `guard` | boolean | An optional flag. If set, and the command returns a non-zero exit code, no further commands for the job will be executed. |
+| `async` | boolean | An optional flag. If set, the command runs asynchronously, in parallel with other future commands. It will be killed when all synchronous commands for a job are finished. |
```

Comment:
What guidance is given to clients as to how much memory that they should configure for their task given that the scheduler can specify async before-commands that co-exist in their process space?